src/lib.rs

```diff
@@ -329,15 +343,76 @@ impl OctocrabBuilder {
             base_url: self
                 .base_url
                 .unwrap_or_else(|| Url::parse(GITHUB_BASE_URL).unwrap()),
+            auth_state: auth_state,
         })
     }
 }
 
+/// A cached API access token (which may be None)
+struct CachedToken(Mutex<Option<String>>);
```

Comment:
This should be a `RwLock`, as you're going to be reading this more than writing to it.

Reply:
Sure